src/engine/strat_engine/setup.rs

```diff
@@ -147,23 +147,18 @@ pub fn get_metadata(pool_uuid: PoolUuid, devnodes: &[PathBuf]) -> EngineResult<O
     // exhausted.
     for &(devnode, ref bda) in
         bdas.iter()
-            .filter(|p| p.1.last_update_time() == most_recent_time) {
-
-        let f = OpenOptions::new().read(true).open(devnode);
-        if f.is_err() {
-            continue;
-        }
-        let mut f = f.expect("f is not err");
-
-        if let Ok(Some(data)) = bda.load_state(&mut f) {
-            let json: serde_json::Result<PoolSave> = serde_json::from_slice(&data);
-            if let Ok(pool) = json {
-                return Ok(Some(pool));
-            } else {
-                continue;
-            }
-        } else {
-            continue;
+            .filter(|&&(_, ref bda)| bda.last_update_time() == most_recent_time) {
+
+        let poolsave = OpenOptions::new()
+            .read(true)
+            .open(devnode)
+            .ok()
+            .and_then(|mut f| bda.load_state(&mut f).ok())
+            .and_then(|opt| opt)
```

Comment:
yes, bda.load_state gives us a `Result<Option<Vec<u8>>>` and this stops the chain if the option is None. I must admit staring at this for a while for a better way to express this but none came to mind. Any suggestions?